skills/core.go

```diff
@@ -36,3 +41,15 @@ type Slot struct {
 	Attrs map[string]interface{}
 	Apps  []string
 }
+
+// Regular expression describing correct identifiers.
+var validName = regexp.MustCompile("^[a-z](:?[a-z0-9-]*[a-z0-9])?$")
```

Comment:
I think this should be: `"^[a-z](?:-?[a-z0-9])*$"`

No double dashes (needs test).